LogAndStream/main.c

```diff
@@ -531,48 +531,49 @@ void main(void)
         if (taskList & TASK_SDLOG_CFG_UPDATE)
         {
             TaskClear(TASK_SDLOG_CFG_UPDATE);
-            if (!docked && !sensing && CheckSdInslot() && GetSdCfgFlag())
+            if (!shimmerStatus.docked && !shimmerStatus.sensing
+                    && CheckSdInslot() && GetSdCfgFlag())
             {
-                configuring = 1;
+                shimmerStatus.configuring = 1;
                 IniReadInfoMem();
                 UpdateSdConfig();
                 SetSdCfgFlag(0);
-                configuring = 0;
+                shimmerStatus.configuring = 0;
             }
         }
 
         if (taskList & TASK_STARTSENSING)
         {
             TaskClear(TASK_STARTSENSING);
-            configuring = 1;
-            if (!sensing && !battCritical)
+            shimmerStatus.configuring = 1;
+            if (!shimmerStatus.sensing && !battCritical)
             {
-                if (newDirFlag && enableSdlog && sdlogReady)
+                if (newDirFlag && shimmerStatus.sdlogCmd && shimmerStatus.sdlogReady)
                 {
                     StartLogging();
-                    //isLogging = 1;
+                    //shimmerStatus.isLogging = 1;
                 }
-                if ((enableSdlog && sdlogReady)
-                        || (enableBtstream && btstreamReady))
+                if ((shimmerStatus.sdlogCmd && shimmerStatus.sdlogReady)
+                        || (shimmerStatus.btstreamCmd && shimmerStatus.btstreamReady))
                 {
                     StartStreaming();
 #if !FW_IS_LOGANDSTREAM
```

Comment:
Is this "!FW_IS_LOGANDSTREAM" check needed here. On SDLog this could cause sync issue.